api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -320,28 +321,36 @@ private void discontinueExistingOrdersIfNecessary(Order order) {
 		    null);
 		boolean isDrugOrderAndHasADrug = DrugOrder.class.isAssignableFrom(getActualType(order))
 		        && ((DrugOrder) order).getDrug() != null;
+		Order orderToBeDiscontinued = null;
 		for (Order activeOrder : orders) {
 			if (!getActualType(order).equals(getActualType(activeOrder))) {
 				continue;
 			}
-			boolean shouldMarkAsDiscontinued = false;
 			//For drug orders, the drug must match if the order has a drug
 			if (isDrugOrderAndHasADrug) {
-				DrugOrder drugOrder1 = (DrugOrder) order;
-				DrugOrder drugOrder2 = (DrugOrder) activeOrder;
-				if (OpenmrsUtil.nullSafeEquals(drugOrder1.getDrug(), drugOrder2.getDrug())) {
-					shouldMarkAsDiscontinued = true;
-				}
-			} else if (activeOrder.getConcept().equals(order.getConcept())) {
-				shouldMarkAsDiscontinued = true;
-			}
-			
-			if (shouldMarkAsDiscontinued) {
-				order.setPreviousOrder(activeOrder);
-				stopOrder(activeOrder, aMomentBefore(order.getDateActivated()));
+				orderToBeDiscontinued = checkDrugOrdersForDiscontinuing((DrugOrder) order, (DrugOrder) activeOrder);
```

Comment:
The same applies for drug orders, if someone calls discontinue a given drug and there is multiple orders with a matching drug and lets say different strengths, then the call is ambiguous too